### PR TITLE
Add primitive geometry conversion

### PR DESCRIPTION
**Pull request by azeey**
# 🎉 New feature

## Summary
This adds conversion utilities for primitive geometries. It also adds basic packaging information to get us started.

## Test it
See README

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: sdformat_mjcf/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[metadata]
+name = sdformat_mjcf
+version = 0.0.1
+author = Addisu Taddese
+author_email = addisu@openrobotics.org
+description = Tool for conversion between SDFormat and MJCF file formats
+long_description = file: README.md
+long_description_content_type = text/markdown
+url = https://github.com/ignitionrobotics/sdformat_mjcf
```
[the remark this answers is not in the document]
Comment (azeey) [reply sdformat_mjcf/setup.cfg]: Nope. Fixed in https://github.com/ignitionrobotics/gz-mujoco/pull/2/commits/77e36e453f6689934094e30abcd961c7b2726d32

File: sdformat_mjcf/src/sdformat_mjcf/converters/geometry.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    elif sdf_geom.mesh_shape():
+        raise RuntimeError("Meshes are not yet supported")
+    else:
+        raise RuntimeError(f"Encountered unsupported shape type {sdf_geom.type()}")
+
+    return geom
+
+
+def add_collision(body, col, pose_resolver=su.pose_resolver):
+    sem_pose = col.semantic_pose()
```
[the remark this answers is not in the document]
Comment (azeey) [reply sdformat_mjcf/src/sdformat_mjcf/converters/geometry.py:98]: Added in https://github.com/ignitionrobotics/gz-mujoco/pull/2/commits/77e36e453f6689934094e30abcd961c7b2726d32

File: sdformat_mjcf/src/sdformat_mjcf/sdf_utils.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+"""Utility functions that aid in converting SDFormat and Gazebo Math types to
+Python"""
+
+import math
+from ignition.math import Pose3d
+
+
+def vec3d_to_list(vec):
+    """Convert a Vector3d object to a list.
+    :param ignition.math.Vector3d vec: Vector3d object to be converted
```
[the remark this answers is not in the document]
Comment (azeey) [reply sdformat_mjcf/src/sdformat_mjcf/sdf_utils.py:24]: Added in https://github.com/ignitionrobotics/gz-mujoco/pull/2/commits/77e36e453f6689934094e30abcd961c7b2726d32

File: sdformat_mjcf/tests/test_add_geometry.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import numpy as np
+from numpy.testing import assert_allclose
+from math import pi
+
+import sdformat as sdf
```
[the remark this answers is not in the document]
Comment (azeey) [reply sdformat_mjcf/tests/test_add_geometry.py:20]: I don't think so. Just like `import numpy as np`, I think it's okay to use a 
short alias in the code, but the package name should be long form.

File: sdformat_mjcf/src/sdformat_mjcf/converters/geometry.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    :param pose_resolver: Function to resolve the pose of a
+    sdformat.SemanticPose object.
+    :return: The newly created MJCF geom.
+    :rtype: mjcf.Element
+    """
+    sem_pose = col.semantic_pose()
+    pose = pose_resolver(sem_pose)
+    geom = add_geometry(body, col.name(), pose, col.geometry())
+    geom.group = VISUAL_GEOM_GROUP
+    return geom
```
Comment (ahcorde) [inline sdformat_mjcf/src/sdformat_mjcf/converters/geometry.py:122]:
```suggestion
def add_visual(body, vis, pose_resolver=su.pose_resolver):
    """
    Converts an SDFormat visual to an MJCF geom and add it to the given
    body. To differentiate Collision geoms from Visual geoms, we assign
    Visual geoms group `VISUAL_GEOM_GROUP`.

    :param mjcf.Element body: The MJCF body to which the geom is added.
    :param sdformat.Visual vis: Visual object to be converted.
    :param pose_resolver: Function to resolve the pose of a
    sdformat.SemanticPose object.
    :return: The newly created MJCF geom.
    :rtype: mjcf.Element
```
    """
    sem_pose = vis.semantic_pose()
    pose = pose_resolver(sem_pose)
    geom = add_geometry(body, vis.name(), pose, vis.geometry())
    geom.group = VISUAL_GEOM_GROUP
    return geom
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: sdformat_mjcf/setup.cfg [hunk 6]
```diff
@@ -0,0 +1,24 @@
+[metadata]
+name = sdformat_mjcf
```
[the remark this answers is not in the document]
Comment (scpeters) [reply sdformat_mjcf/setup.cfg:2]: is this repository going to hold other python packages or just this one?

**Review by ahcorde — CHANGES_REQUESTED** [review]
@j-rivero `python3-sdformat13` is required too

File: sdformat_mjcf/setup.cfg (hunk 6 above)
[the remark this answers is not in the document]
Comment (azeey) [reply sdformat_mjcf/setup.cfg]: My plan was for the `sdformat_mjcf` directory to hold just the `sdformat_mjcf` package, but the parent directory, `gz-mujoco` to hold other python and non-python packages we may create down the road.

File: sdformat_mjcf/src/sdformat_mjcf/converters/geometry.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        geom.size = [cylinder_shape.radius(), cylinder_shape.length() / 2.0]
+    elif sdf_geom.ellipsoid_shape():
+        ellipsoid_shape = sdf_geom.ellipsoid_shape()
+        geom.type = "ellipsoid"
+        geom.size = su.vec3d_to_list(ellipsoid_shape.radii())
+    elif sdf_geom.plane_shape():
+        plane_shape = sdf_geom.plane_shape()
+        geom.type = "plane"
+        # The third element of size defines the spacing between square grid
+        # lines for rendering.
```
[the remark this answers is not in the document]
Comment (azeey) [reply sdformat_mjcf/src/sdformat_mjcf/converters/geometry.py]: The documentation in https://mujoco.readthedocs.io/en/latest/XMLreference.html#body-geom doesn't say, but I gave it a try and what I found is that if the first two elements are 0, then the third element is completely ignored. If one of the two elements are nonzero, the third element has to be positive. Setting it to zero won't break the conversion script, but the Mujoco simulator fails to load the file. Btw, this is only visible in Wireframe mode enabled under Rendering->OpenGL Effects.

File: sdformat_mjcf/src/sdformat_mjcf/converters/geometry.py (hunk 5 above)
Comment (azeey) [reply sdformat_mjcf/src/sdformat_mjcf/converters/geometry.py]: Done in https://github.com/gazebosim/gz-mujoco/pull/2/commits/5af05fddd1f4693e261da624f2a30c37a638db91

**Pull request merged**